server/cluster/cluster.go

```diff
@@ -606,6 +607,33 @@ func (c *RaftCluster) HandleStoreHeartbeat(stats *pdpb.StoreStats) error {
 	return nil
 }
 
+// processBucketHeartbeat update the bucket information.
+func (c *RaftCluster) processBucketHeartbeat(buckets *metapb.Buckets) error {
+	c.RLock()
+	region := c.core.GetRegion(buckets.GetRegionId())
+	c.RUnlock()
```

Comment:
Why we need lock? It seems `region := c.core.GetRegion(buckets.GetRegionId())` is thread safe

Reply:
fix